packages/seed/src/types.ts

```diff
@@ -0,0 +1,76 @@
+import * as auth from 'aws-amplify/auth';
+
+export type AuthSignUp = {
+  signInAfterCreation: boolean;
+  username: string;
+  userAttributes?: StandardUserAttributes;
+} & (PasswordSignInFlow | MfaSignUpFlow | MfaWithTotpSignUpFlow);
+
+export type AuthUser = {
+  username: string;
+} & (PasswordSignInFlow | MfaSignInFlow);
+
+export type AuthUserGroupInput = {
+  username: string;
+};
+
+export type AuthOutputs = {
+  signInFlow: 'Password' | 'MFA';
+  username: string;
+};
+
+export type ChallengeResponse = {
+  challengeResponse: string;
+};
+
+export type PasswordSignInFlow = {
+  signInFlow: 'Password';
+  password: string;
```

Comment:
why don't we do this?

```suggestion
  password: string | BackendSecret;
```

@sobolk did we discuss allowing developers to just pass the secrets reference?

Reply:
That would have to be different `BackendSecret`. 
The existing one is tightly coupled to `Construct` , see https://github.com/aws-amplify/amplify-backend/blob/aec5e08590195d9eb9a0d73be06fe36fb1a09856/packages/plugin-types/src/backend_secret_resolver.ts#L10 .

We could potentially invent `SeedSecret` as a type/class.
And then have `get/setSecret` absorbed into it.

But the thing is. We agreed to provide APIs to deal with passwords using secure storage, but we don't aim to secure in memory runtime access.